### PR TITLE
New: first work on localization

### LINKED ISSUE #269: Localization in Cappuccino Apps
May I know will it be possible to have localization support for cappuccino?

### PR DESCRIPTION
**Pull request by Dogild**
This pull request add the following new features :

- Added the protocol CPBundleDelegate
- Added the methods `CPLocalizedString(key, comment)` `CPLocalizedStringFromTable(key, table, comment)` `CPCopyLocalizedStringFromTableInBundle(key, table, bundle, comment)`
- Added the method `CPBundle -localizedStringForKey:value:table:`
- Added base localization and interface builder localization for a xib (can only use once at a time)
- Added methods `CPBundle pathForResource:ofType:`, `CPBundle pathForResource:ofType:inDirectory:`, `CPBundle pathForResource:ofType:inDirectory:forLocalization:`
- Info.plist needs to be configure with the keys `CPBundleLocalizableStrings`, `CPBundleTypeOfLocalization ` and `CPBundleDefaultLanguage ` (as javascript was not made to list the files of a computer)

Test app in Tests/Manual/CPLocalizationTest
UnitTest in Tests/Foundation/CPBundleTest

Fixed #269

### CODE REVIEW

File: AppKit/Cib/_CPLocalizableString.j [hunk 1]
```diff
@@ -0,0 +1,66 @@
+/*
+ * _CPLocalizableString.j
+ * AppKit
+ *
+ * Created by Alexandre Wilhelm.
+ * Copyright 2008, 280 North, Inc.
```
Comment (aparajita) [inline AppKit/Cib/_CPLocalizableString.j]: If this is a new file, it should be Copyright 2015, The Cappuccino Project

**Comment by Dogild** [comment]
Progress : methods `CPLocalizedString`, `CPLocalizedStringFromTable` and `CPCopyLocalizedStringFromTableInBundle` works as expected (with and without the context).

Only difference with cocoa, we need to declare in the Info.plist which localizable files (key -> CPLocalizableStrings) to load. The application will take the navigator language to know which language to import

**Comment by Dogild** [comment]
Refs #269

**Comment by Dogild** [comment]
Support for xibs has been added. Right now we are slightly different than Cocoa.

We need to add in the Info.plist the strings you need to load, for instance :
'''
    <key>CPBundleLocalizableStrings</key>
    <array>
        <string>Localizable.strings</string>
        <string>SecondLocalizable.strings</string>
        <string>MainMenu.strings</string>
    </array>
'''

Right now, Cappuccino supports Based Localization or Interface Builder Localization, not possible to have both together. Options in the Info.plist :

'''
    <key>CPBundleTypeOfLocalization</key>
    <string>CPBundleBaseLocalizationType</string>
'''

You can force the language of the application in the Info.plist :

'''
    <key>CPBundleDefaultLanguage</key>
    <string>CPBundleDefaultBrowserLanguage</string> or <string>en</string> 
'''

@aparajita, @primalmotion what do you think about that ?

**Comment by Dogild** [comment]
We have just integrated this branch on our master. We will know soon if there is a regression or not.

**Comment by Dogild** [comment]
Works here, no regression found.

**Pull request merged**